src/plugins/intel_gpu/include/intel_gpu/graph/serialization/binary_buffer.hpp

```diff
@@ -36,32 +37,43 @@ class BinaryOutputBuffer : public OutputBuffer<BinaryOutputBuffer> {
 class BinaryInputBuffer : public InputBuffer<BinaryInputBuffer> {
 public:
     BinaryInputBuffer(std::istream& stream, engine& engine)
-    : InputBuffer(this, engine), stream(stream), _impl_params(nullptr) {}
+    : InputBuffer(this, engine), _stream(stream), _impl_params(nullptr), _num_networks(0), _stream_id(0) {}
 
     void read(void* const data, std::streamsize size) {
-        auto const read_size = stream.rdbuf()->sgetn(reinterpret_cast<char*>(data), size);
+        auto const read_size = _stream.rdbuf()->sgetn(reinterpret_cast<char*>(data), size);
         OPENVINO_ASSERT(read_size == size,
             "[GPU] Failed to read " + std::to_string(size) + " bytes from stream! Read " + std::to_string(read_size));
     }
 
     void setKernelImplParams(void* impl_params) { _impl_params = impl_params; }
     void* getKernelImplParams() const { return _impl_params; }
-    void addConstData(const std::string& prim_id, const std::shared_ptr<memory> mem_ptr) {
-        OPENVINO_ASSERT(_const_data_map.find(prim_id) == _const_data_map.end(), "[GPU] duplicated primitive id " + prim_id);
-        _const_data_map[prim_id] = mem_ptr;
+    void addConstData(const uint32_t net_id, const std::string& prim_id, const std::shared_ptr<memory> mem_ptr) {
+        while (_const_data_map.size() <= net_id) {
+            _const_data_map.emplace_back(std::unordered_map<std::string, std::shared_ptr<memory>>());
+        }
+        OPENVINO_ASSERT(_const_data_map[net_id].find(prim_id) == _const_data_map[net_id].end(), "[GPU] duplicated primitive id " + prim_id);
+        _const_data_map[net_id][prim_id] = mem_ptr;
     }
-    std::shared_ptr<memory> getConstData(const std::string& prim_id) {
-        OPENVINO_ASSERT(_const_data_map.find(prim_id) != _const_data_map.end(), "[GPU] Not found primitive id " + prim_id);
-        return _const_data_map[prim_id];
+    std::shared_ptr<memory> getConstData(const uint32_t net_id, const std::string& prim_id) {
+        OPENVINO_ASSERT(_const_data_map[net_id].find(prim_id) != _const_data_map[net_id].end(), "[GPU] Not found primitive id " + prim_id);
+        return _const_data_map[net_id][prim_id];
     }
 
-    std::streampos tellg() { return stream.tellg(); }
-    void seekg(std::streampos pos) { stream.seekg(pos); }
+    std::streampos tellg() { return _stream.tellg(); }
+    void seekg(std::streampos pos) { _stream.seekg(pos); }
+
+    void new_network_added() { _num_networks += 1; }
+    int get_num_networks() { return _num_networks; }
```

Comment:
nit: It seems better to specify const keyword for "read-only" function.
```suggestion
    int get_num_networks() const { return _num_networks; }
```

Reply:
I added "const" as you commented. Thank you.